src/nunit.analyzers.tests/NonTestMethodAccessibilityLevel/NonTestMethodAccessibilityLevelAnalyzerTests.cs

```diff
@@ -81,5 +81,96 @@ public void AnalyzeWhenNonTestMethodIsPublicOrInternal(string modifiers)
         {modifiers} void ↓AssertMethod() {{ }}");
             AnalyzerAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
         }
+
+        [TestCaseSource(nameof(TestMethodRelatedAttributes))]
+        public void AnalyzeWhenTestMethodIsPublicAndOverridden(string attribute)
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing($@"
+        public class BaseTestClass
+        {{
+            [{attribute}]
+            public virtual void TestMethod() {{ }}
+        }}
+
+        public sealed class DerivedTestClass : BaseTestClass
+        {{
+            public override void TestMethod() {{ }}
+
+            [Test]
+            public void OtherTestMethod() {{ }}
+        }}
+        ");
+
+            AnalyzerAssert.Valid<NonTestMethodAccessibilityLevelAnalyzer>(testCode);
+        }
+
+        [Test]
+        public void AnalyzeConstructor()
```

Comment:
Is this just a testcase that we forgot? (I guess that this is handled by the existing check for `MethodKind.Ordinary`)?

Reply:
Yes, I wanted to make sure that we don't trip up when the new nunit support for instance-per-test comes out where SetUp/TearDown is replaced with contructor/Dispose to better support nullable fields.

Reply:
God idea. I had not thought about this.